libs/openFrameworks/app/ofAppBaseWindow.h

```diff
@@ -18,111 +19,118 @@ struct _XDisplay;
 typedef struct _XDisplay Display;
 #endif
 
-class ofAppBaseWindow{
+class ofAppBaseWindow {
 public:
+	ofAppBaseWindow() { }
+	virtual ~ofAppBaseWindow() { }
 
-	ofAppBaseWindow(){}
-	virtual ~ofAppBaseWindow(){}
-
-	virtual void setup(const ofWindowSettings & settings)=0;
-	virtual void update()=0;
-	virtual void draw()=0;
-	virtual bool getWindowShouldClose(){
+	virtual void setup(const ofWindowSettings & settings) = 0;
+	virtual void update() = 0;
+	virtual void draw() = 0;
+	virtual bool getWindowShouldClose() {
 		return false;
 	}
-	virtual void setWindowShouldClose(){}
-	virtual void close(){}
+	virtual void setWindowShouldClose() { }
+	virtual void close() { }
 	virtual ofCoreEvents & events() = 0;
 	virtual std::shared_ptr<ofBaseRenderer> & renderer() = 0;
 
-	virtual void hideCursor(){}
-	virtual void showCursor(){}
+	virtual void hideCursor() { }
+	virtual void showCursor() { }
+
+	virtual void setWindowPosition(int x, int y) { }
+	virtual void setWindowSize(int w, int h) { }
+	virtual void setWindowMinimumSize(int w, int h) { }
+	virtual void setWindowMaximumSize(int w, int h) { }
+	virtual void setWindowAspectRatio(int horizontal, int vertical) { }
+	virtual void setWindowShape(int w, int h) { }
 
-	virtual void setWindowPosition(int x, int y){}
-	virtual void setWindowShape(int w, int h){}
+	std::optional<glm::vec2> minimumWindowSize;
+	std::optional<glm::vec2> maximumWindowSize;
+	std::optional<glm::vec2> windowAspectRatio;
```

Comment:
can we make these protected ?

Reply:
certainly! however in relation to the GLUT implementation (which is currently commented out) if the only window typer supporting these limits is GLFW, they could also be moved into private ofAppGLFWWindow?

(at the same time perhaps confirm it is not a priority to support GLUT? (who's callbacks are statics; makes things a bit different)

Reply:
yeah - we should do a PR to remove all references to GLUT at this point. 
🙂